### ISSUE TITLE
Consider using AccessLevelHelper.GetEffectiveAccessibility for SA1623/SA1624

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#1811]
Currently the implementation of SA1623/SA1624 relies on a locally-implemented method `IsPubliclyVisible`. The helper method `AccessLevelHelper.GetEffectiveAccessibility` implements a more precise and accurate version of this functionality. If possible, the implementation of SA1623/SA1624 should be update to use it.

Marked as a :bug: because the implementation of `IsPubliclyVisible` currently looks for the `private` modifier for classes, when it should probably be looking for the `internal` modifier.

**Comment by vweijsters** [comment]
It should be looking for internal when the accessibility of the type (and all of its parents) is public, it should be looking for private when the accessibility of the type (or one of its parents) is internal.

**Comment by vweijsters** [comment]
Grabbing this
